cerberus-audit-logger-athena/src/test/java/com/nike/cerberus/audit/logger/listener/AthenaLoggingEventListenerTest.java

```diff
@@ -0,0 +1,318 @@
+package com.nike.cerberus.audit.logger.listener;
+
+import com.nike.cerberus.event.AuditableEvent;
+import com.nike.cerberus.event.AuditableEventContext;
+import java.time.OffsetDateTime;
+import java.util.Optional;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.MockitoAnnotations;
+import org.slf4j.Logger;
+
+public class AthenaLoggingEventListenerTest {
+
+  @Mock private Logger auditLogger;
+
+  @InjectMocks private AthenaLoggingEventListener athenaLoggingEventListener;
+
+  @Before
+  public void setup() {
+    MockitoAnnotations.initMocks(this);
+  }
+
+  @Test
+  public void testIfAuditableContextIsMissingInAuditEvent() {
+    AuditableEvent auditableEvent = Mockito.mock(AuditableEvent.class);
+    athenaLoggingEventListener.onApplicationEvent(auditableEvent);
+    Mockito.verify(auditLogger, Mockito.never()).info(Mockito.anyString());
+  }
+
+  @Test
+  public void testIfAuditableContextIsPresentInAuditEvent() {
+    AuditableEvent auditableEvent = Mockito.mock(AuditableEvent.class);
+    AuditableEventContext auditableEventContext = Mockito.mock(AuditableEventContext.class);
+    Mockito.when(auditableEventContext.getTraceId()).thenReturn("traceId");
+    Mockito.when(auditableEventContext.getTimestamp())
+        .thenReturn(OffsetDateTime.parse("2007-12-03T10:15:30+01:00"));
+    Mockito.when(auditableEventContext.getIpAddress()).thenReturn("ipAddress");
+    Mockito.when(auditableEventContext.getXForwardedFor()).thenReturn("xforwarder");
+    Mockito.when(auditableEventContext.getVersion()).thenReturn("version");
+    Mockito.when(auditableEventContext.getClientVersion()).thenReturn("clientVersion");
+    Mockito.when(auditableEventContext.getMethod()).thenReturn("post");
+    Mockito.when(auditableEventContext.getStatusCode()).thenReturn(200);
+    Mockito.when(auditableEventContext.getPath()).thenReturn("/path");
+    Mockito.when(auditableEventContext.getAction()).thenReturn("action");
+    Mockito.when(auditableEventContext.isSuccess()).thenReturn(true);
+    Mockito.when(auditableEventContext.getEventName()).thenReturn("eventName");
+    Mockito.when(auditableEventContext.getOriginatingClass()).thenReturn("originatingClass");
+    Mockito.when(auditableEventContext.getPrincipalName()).thenReturn("pricinpleName");
+    Mockito.when(auditableEvent.getAuditableEventContext()).thenReturn(auditableEventContext);
+    Mockito.when(auditableEventContext.getPrincipalAsCerberusPrincipal())
+        .thenReturn(Optional.empty());
+    athenaLoggingEventListener.onApplicationEvent(auditableEvent);
+    Mockito.verify(auditLogger).info(Mockito.anyString());
+  }
+
+  @Test(expected = NullPointerException.class)
```

Comment:
All of these tests are testing if certain items are missing in context. Can we discuss what this test and the ones below are testing? It looks like you are just adding more information to the auditableEventContext, but necessarily testing functionality.

Reply:
I'm trying to test insertion null values for each key that we are putting in Immutable hashmap and expecting NullPointerException having property in the message , but when comes to event_timestamp without checking null we are trying to format it so we are getting NullPointerException, so could not test this  like the way we are
testing other properties.